### ISSUE TITLE
Official name

### ISSUE THREAD

**Issue by edwinksl** [Microsoft/WSL#1013]
I am confused by the plethora of names this project has acquired:
1. Windows Subsystem for Linux (WSL)
2. Bash on Windows
3. Ubuntu on Windows
4. Bash on Ubuntu on Windows

Which of these names is official? Which one should we use when discussing the project? Are there authoritative sources that shed light on this?

**Comment by fpqc** [comment]
Not sure because I'm not a Microsoft Employee, but I believe that 1 is the name for the pair of pico drivers, lxcore.sys and lxss.sys, the session manager service, and the pico process collection that makes up the "Linux Instances"

The other three are pretty easily mutable, namely, the system allows you to run, say, "zsh on Arch on Windows".  Right now, the WSL does not support all of the features used by Arch's package manager, pacman,  but there is no code tying you to Ubuntu.

You can for example replace Ubuntu with Debian (since they share a package management system) without much hassle at all, so you could have "FiSH on Debian on Windows".  So I would say that descriptions 2-4 are "descriptions of applications of WSL".

**Comment by SRGOM** [comment]
WSL is the most unique of them so I'd like if everyone uses this and it picks up. Stackoverflow tag is WSL for what its worth.

**Comment by iz0eyj** [comment]
WSL, and please (Microsoft) kill the name "Bash on Windows" :)

**Comment by SRGOM** [comment]
Are you offering to pay to license the "Linux" trademark ;)? (assuming you want the name linux somewhere in there.... )

**Comment by iz0eyj** [comment]
No, "Linux" cannot be used, simply WSL.
Ig someone asks... Windows Subsystem for... bah... Liberty? :)

**Comment by fpqc** [comment]
@SRGOM The rules for the Linux trademark are pretty lax:

https://www.linuxmark.org/programs/legal/trademark/attribution

"At the end of your web page, advertisement, publication or media broadcast, include the following text in a legible font and size: Linux® is the registered trademark of Linus Torvalds in the U.S. and other countries."

Also read:

https://en.wikipedia.org/wiki/Linux_Mark_Institute

Basically, Microsoft would not need to do anything substantial to be in compliance, if it would need to do anything at all.

**Comment by SRGOM** [comment]
@iz0eyj 100% behind "Liberty" in name! :)

@fpqc: What about adding in Windows?

**Comment by mobluse** [comment]
#WSL also means World Surf League on Twitter. #BashOnWindows is more unique on Twitter.

**Comment by aseering** [comment]
I'm a little surprised no one has pointed this out yet -- before "bash" was the name of a shell, it was a regular English word.  According to the dictionary definition, to bash on something is to physically or verbally abuse it.  For example, "free-software activists love to bash on Windows."  (Also, don't look up the word's common slang definition in Urban Dictionary; the example sentence above gains a whole new meaning...)

Irony can be a wonderful thing.  But everyone kind of has to realize that your choice of name is, in fact, ironic for the wonderfulness to work :-)

**Comment by SRGOM** [comment]
@mobluse I don't remember the last time I got anything technically useful on twitter. Also, twitter is censoring anti-Hillary accounts for absolutely no wrongdoing- https://twitter.com/PrisonPlanet/status/771650647074545665 (Sorry I know this is the wrong place to post this but we're one step from Gestapo, there's a twitter ban on talking about Hillary's corruption/health)

**Comment by Zenexer** [comment]
I've been calling it Lxss (or lxss).  People seem to recognize that better than WSL, since it shows up all over the place.  "_____ on _____ [on _____]" is too cumbersome for casual conversation.

**Comment by danielgjackson** [comment]
Of course it's actually a "Linux Subsystem for Windows".  ("Wine" is more of a Windows Subsystem for Linux...)

**Comment by aseering** [comment]
I've personally seen better recognition for "WSL"; I haven't personally seen "LXSS" used all that often.  Too many hard consonants?

"Windows Subsystem" is a technical term in the Windows ecosystem; this thing is "one of those, for Linux".

**Comment by danielgjackson** [comment]
...yes, but while I might know that:

1. previous Windiws subsystems are often known with the environment's name first, like the "POSIX Subsystem" and the "OS/2 Subsystem"

2. "<X> for <OS>" is a common pattern for naming software to run on the specified operating system

...I think it is reasonable that some people could initially be confused by that name!

**Comment by mufunyo** [comment]
In casual conversation I either call it "Linux for Windows", or "(the) Linux subsystem" if Windows is already the established conversation subject.

**Comment by benhillis** [comment]
Windows Subsystem for Linux is the preferred nomenclature, dude...
